From 414d309e17bfb38d5cba3d5cd067541a809450d9 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Tue, 19 Aug 2014 15:26:10 -0400
Subject: [tools] perf/record: Improve write_output error message

Message-id: <1408462094-14194-202-git-send-email-jolsa@redhat.com>
Patchwork-id: 88165
O-Subject: [PATCH RHEL7.1 BZ1131394 201/325] perf record: Improve write_output error message
Bugzilla: 1131394
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1131394
https://bugzilla.redhat.com/show_bug.cgi?id=1131394

upstream
========
commit 4f624685f92719565981eb6f8d9195bb578522a3
Author: Adrian Hunter <adrian.hunter@intel.com>
Date: Fri Oct 18 15:29:00 2013 +0300

description
===========
Improve the error message from write_output() to say what failed to
write and give the error number.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/builtin-record.c b/tools/perf/builtin-record.c
index e767e01..7ce0a1e 100644
--- a/tools/perf/builtin-record.c
+++ b/tools/perf/builtin-record.c
@@ -88,7 +88,7 @@ static int write_output(struct perf_record *rec, void *buf, size_t size)
   int ret = write(rec->output, buf, size);
 
   if (ret < 0) {
-   pr_err("failed to write\n");
+   pr_err("failed to write perf data, error: %m\n");
    return -1;
   }
 
-- 
1.7.1